lcls_live/datamaps/klystron.py

```diff
@@ -266,9 +265,13 @@ def klystron_pvinfo(sector, station, beamcode=1):
         description += ' for special feedback'
         base =  f'KLYS:LI{sector}:{station}1'     # Normal base
         phase = f'ACCL:LI24:{station}00:KLY_PDES' # Readback
+        if beamcode == 2:
+            phase += ':SETDATA_1' # Suggested by FJD
```

Comment:
I'd use :GETDATA_1 instead of :SETDATA_1 (both probably work fine 99% of the time, but :GETDATA_1 is really the 'readback PV')

Additionally, you should append :GETDATA_0 onto the end in the beamcode == 1 case.  Otherwise, I think you _might_ get beamcode 2 phases sometimes.

The drawback of GETDATA is that it apparently wasn't getting archived (although I've just added it to the archiver), so maybe its worth it to keep using SETDATA.  Up to you.

Reply:
ok done